src/secret_sharing/malicious_replicated.rs

```diff
@@ -140,38 +146,42 @@ impl<F: Field> Mul<F> for MaliciousReplicated<F> {
     }
 }
 
+#[async_trait]
 impl<F: Field> Downgrade for MaliciousReplicated<F> {
     type Target = Replicated<F>;
-    fn downgrade(self) -> UnauthorizedDowngradeWrapper<Self::Target> {
+    async fn downgrade(self) -> UnauthorizedDowngradeWrapper<Self::Target> {
         UnauthorizedDowngradeWrapper(self.x)
     }
 }
 
+#[async_trait]
 impl<T, U> Downgrade for (T, U)
 where
-    T: Downgrade,
-    U: Downgrade,
+    T: Downgrade + Send,
+    U: Downgrade + Send,
 {
-    type Target = (<T as Downgrade>::Target, <U as Downgrade>::Target);
-    fn downgrade(self) -> UnauthorizedDowngradeWrapper<Self::Target> {
-        UnauthorizedDowngradeWrapper((
-            self.0.downgrade().access_without_downgrade(),
-            self.1.downgrade().access_without_downgrade(),
-        ))
+    type Target = (<T>::Target, <U>::Target);
+    async fn downgrade(self) -> UnauthorizedDowngradeWrapper<Self::Target> {
+        let one = self.0.downgrade();
+        let two = self.1.downgrade();
+        let out = join(one, two).await;
+        let output = (out.0.access_without_downgrade(), out.1.access_without_downgrade());
+        UnauthorizedDowngradeWrapper(output)
```

Comment:
@martinthomson error is following
raw_ipa::secret_sharing::malicious_replicated
pub struct UnauthorizedDowngradeWrapper<T>
[1 implementation](vscode-file://vscode-app/private/var/folders/_r/s7hk62fn5c16yfvqlg3whg400000gn/T/AppTranslocation/71EC7F17-8378-470B-819D-C269580AE45A/d/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)

future cannot be sent between threads safely
within `impl futures::Future<Output = UnauthorizedDowngradeWrapper<(<T as Downgrade>::Target, <U as Downgrade>::Target)>>`, the trait `std::marker::Send` is not implemented for `<T as Downgrade>::Target`
required for the cast from `impl futures::Future<Output = UnauthorizedDowngradeWrapper<(<T as Downgrade>::Target, <U as Downgrade>::Target)>>` to the object type `dyn futures::Future<Output = UnauthorizedDowngradeWrapper<(<T as Downgrade>::Target, <U as Downgrade>::Target)>> + std::marker::Send`